src/uthash.h

```diff
@@ -342,17 +356,17 @@ do {
         _hd_hh_del = &((delptr)->hh);                                            \
         if ((delptr) == ELMT_FROM_HH((head)->hh.tbl,(head)->hh.tbl->tail)) {     \
             (head)->hh.tbl->tail =                                               \
-                (UT_hash_handle*)((ptrdiff_t)((delptr)->hh.prev) +               \
+                (UT_hash_handle*)((uthash_ptrdiff_t)((delptr)->hh.prev) +        \
                 (head)->hh.tbl->hho);                                            \
         }                                                                        \
         if ((delptr)->hh.prev != NULL) {                                         \
-            ((UT_hash_handle*)((ptrdiff_t)((delptr)->hh.prev) +                  \
+            ((UT_hash_handle*)((uthash_ptrdiff_t)((delptr)->hh.prev) +           \
                     (head)->hh.tbl->hho))->next = (delptr)->hh.next;             \
         } else {                                                                 \
             DECLTYPE_ASSIGN(head,(delptr)->hh.next);                             \
         }                                                                        \
         if (_hd_hh_del->next != NULL) {                                          \
-            ((UT_hash_handle*)((ptrdiff_t)_hd_hh_del->next +                     \
+            ((UT_hash_handle*)((uthash_ptrdiff_t)_hd_hh_del->next +              \
```

Comment:
My kneejerk reaction here is "why on earth does this cast exist in the first place"; I'd like to dig a little deeper here and figure out whether we can get away with a simpler type, such as `unsigned` or `void*` or `char*` or no-cast-at-all.

I'm inclined to accept the uthash_strlen, uthash_memcmp stuff immediately and defer the ptrdiff_t stuff to another PR; what do you think?